### PR TITLE
feat: submit pushed_at to TMC when running in PR GHA

### PR DESCRIPTION
**Pull request by wmalik**
## What this PR does / why we need it:

* When cloud-sync is running in the context of a pull_request GHA event, the `event.pull_request.head.repo.pushed_at` is sent to TMC in the `review_request.pushed_at` field.

## Does this PR introduce a user-facing change?
```
No.

```

### CODE REVIEW

File: cmd/terramate/e2etests/cloud/run_cloud_sync_preview_test.go [hunk 1]
```diff
@@ -78,12 +87,31 @@ func TestCLIRunWithCloudSyncPreview(t *testing.T) {
...
+						Platform:       "github",
+						Repository:     "terramate.io/terramate-io/dummy-repo.git",
+						CommitSHA:      "6dcb09b5b57875f334f61aebed695e2e4193db5e",
+						Number:         1347,
+						Title:          "Amazing new feature",
+						Description:    "Please pull these awesome changes in!",
+						URL:            "https://github.com/octocat/Hello-World/pull/1347",
+						Labels:         []cloud.Label{{Name: "bug", Color: "f29513", Description: "Something isn't working"}},
+						Status:         "open",
+						ReviewRequired: true,
```
Comment (sixstone-qq) [inline cmd/terramate/e2etests/cloud/run_cloud_sync_preview_test.go:100]: This clashes with https://github.com/terramate-io/terramate/pull/1525 . Have you rebase from main?

Comment (wmalik) [reply cmd/terramate/e2etests/cloud/run_cloud_sync_preview_test.go:100]: Not yet, will address this and ping you again.

**Review by sixstone-qq — DISMISSED** [review]
LGTM

File: cmd/terramate/cli/cloud.go [hunk 2]
```diff
@@ -538,13 +538,16 @@ func (c *cli) detectCloudMetadata() {
...
 	var prNumber int
+	var pushedAtFromEvent *time.Time
 	prFromEvent, err := tmgithub.GetEventPR()
 	if err != nil {
 		logger.Debug().Err(err).Msg("unable to get pull_request details from GITHUB_EVENT_PATH")
 	} else {
 		logger.Debug().Err(err).Msg("got pull_request details from GITHUB_EVENT_PATH")
 		c.cloud.run.prFromGHAEvent = prFromEvent
 		prNumber = prFromEvent.GetNumber()
+		pushedAtTimestamp := prFromEvent.GetHead().GetRepo().GetPushedAt()
```
Comment (i4ki) [inline cmd/terramate/cli/cloud.go:549]: in this chaining, is it guaranteed that this path traversal will not panic if any of the intermediate fields do not exist?

Comment (wmalik) [reply cmd/terramate/cli/cloud.go]: It was not safe, I just added nil checks.

File: cmd/terramate/cli/cloud.go [hunk 3]
```diff
@@ -885,6 +891,12 @@ func (c *cli) newReviewRequest(pull *github.PullRequest, reviews []*github.PullR
 		UpdatedAt:      pullUpdatedAt.GetTime(),
 	}
 
+	prFromEvent := c.cloud.run.prFromGHAEvent
+	if prFromEvent.GetHead() != nil && prFromEvent.GetHead().GetRepo() != nil {
+		pushedAt := prFromEvent.GetHead().GetRepo().GetPushedAt()
```
[the remark this answers is not in the document]
Comment (wmalik) [reply cmd/terramate/cli/cloud.go:896]: No, it will always be a github.Timestamp (not a pointer)

**Pull request merged**